ext/src/zpages/tracez_data_aggregator.cc

```diff
@@ -0,0 +1,112 @@
+#include "opentelemetry/ext/zpages/tracez_data_aggregator.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace zpages
+{
+
+TracezDataAggregator::TracezDataAggregator(std::shared_ptr<TracezSpanProcessor> span_processor)
+{
+  tracez_span_processor_ = span_processor;
+}
+
+const std::map<std::string, std::unique_ptr<AggregatedInformation>>& TracezDataAggregator:: GetAggregatedData()
+{
+  AggregateSpans();
+  return aggregated_data_;
+}
+
+LatencyBoundaryName TracezDataAggregator::GetLatencyBoundary(opentelemetry::sdk::trace::Recordable* recordable)
+{
+  auto recordable_duration = recordable->GetDuration();
+  for(int boundary = 0; boundary < kNumberOfLatencyBoundaries-1; boundary++)
+  {
+    auto lower_limit = kLatencyBoundaries[boundary];
+    auto upper_limit = kLatencyBoundaries[boundary+1];
+    if(lower_limit <= recordable_duration && recordable_duration < upper_limit)return (LatencyBoundaryName)boundary;
```

Comment:
A possible optimization is to have only upper limit, and then compare it always with the upper limit (because the lower limit is always the upper limit of the previous bucket).